File: fix-images.js
const fs = require('fs');
const glob = require('crypto').randomBytes; // well I don't need glob, I can just read dirs

const files = [
  "app/about/page.tsx",
  "app/blogs/page.tsx",
  "app/components/about.tsx",
  "app/components/gallery.tsx",
  "app/components/service1.tsx",
  "app/photography/page.tsx",
  "app/portfolio/page.tsx",
  "app/service1/page.tsx",
  "app/service2/page.tsx",
  "app/service3/page.tsx",
  "app/service4/page.tsx",
  "app/service5/page.tsx",
  "app/service6/page.tsx",
  "app/service7/page.tsx",
  "app/service8/page.tsx",
  "app/service9/page.tsx",
  "app/service10/page.tsx"
];

let replacedCount = 0;

for (const file of files) {
  if (!fs.existsSync(file)) continue;

  let content = fs.readFileSync(file, 'utf8');
  let changed = false;

  // Ensure Import
  if (content.includes('<img') && !content.includes('next/image')) {
    if (content.includes('import React')) {
      content = content.replace(/import React(.*?)\n/, 'import React$1\nimport Image from "next/image";\n');
    } else {
      content = 'import Image from "next/image";\n' + content;
    }
    changed = true;
  }

  // 1. the manual style one in service/about parallax ParallaxImg
  if (content.includes('width: "100%",\n            height: "124%",')) {
      content = content.replace(
        /<img\s+src=\{src\}\s+alt=\{alt\}\s+style=\{\{(.*?)\}\}\s*\/>/gs, 
        '<Image src={src} alt={alt} width={1200} height={1600} style={{$1}} />'
      );
      changed = true;
  }

  // 2. Generic <img src={...} className="..." />
  // Careful not to match if it spans weirdly, we match simple tags
  const imgRegex = /<img\s+([^>]*?)(\/?)>/g;
  content = content.replace(imgRegex, (match, attrs, selfClose) => {
    if (match.includes('width={') || match.includes('fill')) return match; // Already handled or ignored?
    // We can just add fill and sizes if they don't have style width/height inside
    // Because sometimes img is in a container. Let's just use width={1200} height={1200} or fill
    // If it has object-cover, it's safer to use fill if the parent has relative/overflow-hidden.
    // Wait, let's just make it <Image width={1200} height={1200} ... > to be safe if fill requires parent relative.
    
    // We already handled the parallax `img` in Service files. Let's handle the rest by adding width/height.
    if (!match.includes('width=') && !match.includes('fill')) {
        return `<Image width={1200} height={1200} ${attrs}${selfClose ? '/' : ''}>`;
    }
    return match;
  });

  if (content !== fs.readFileSync(file, 'utf8')) {
    fs.writeFileSync(file, content);
    console.log(`Updated ${file}`);
    replacedCount++;
  }
}

console.log('Done, modified', replacedCount, 'files.');

File: fix-images.py
import os
import re

files_to_check = [
    "app/about/page.tsx",
    "app/blogs/page.tsx",
    "app/components/about.tsx",
    "app/components/gallery.tsx",
    "app/components/service1.tsx",
    "app/photography/page.tsx",
    "app/photography/[id]/page.tsx",
    "app/portfolio/page.tsx",
    "app/service1/page.tsx",
    "app/service2/page.tsx",
    "app/service3/page.tsx",
    "app/service4/page.tsx",
    "app/service5/page.tsx",
    "app/service6/page.tsx",
    "app/service7/page.tsx",
    "app/service8/page.tsx",
    "app/service9/page.tsx",
    "app/service10/page.tsx"
]

def fix_file(filepath):
    if not os.path.exists(filepath):
        return
        
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    if '<img' not in content:
        return
        
    original_content = content
    
    # 1. Add import Image from "next/image"
    if 'next/image' not in content:
        if 'import React' in content:
            content = re.sub(r'(import React.*?)\n', r'\1\nimport Image from "next/image";\n', content, count=1)
        elif 'import { ' in content:
            content = 'import Image from "next/image";\n' + content
        else:
            content = 'import Image from "next/image";\n' + content

    # Replace specific parallax <img> in service*/page.tsx
    content = re.sub(
        r'<img\s+src=\{src\}\s+alt=\{alt\}\s+style=\{\{(.*?)\}\}\s*\/>',
        r'<Image src={src} alt={alt} width={1200} height={1600} style={{\1}} />',
        content,
        flags=re.DOTALL
    )

    # Replace other <img ... /> generic instances by adding width and height to be safe
    # Using a simple logic: if <img doesn't have width= or <Image it gets replaced to Image width height
    def sub_img(match):
        attrs = match.group(1)
        if 'width=' in attrs or 'fill' in attrs:
            if match.group(0).startswith('<img'):
                return f'<Image {attrs}>'
            return match.group(0)
        
        # Determine some reasonable defaults
        return f'<Image width={{1600}} height={{1600}} {attrs}>'

    content = re.sub(r'<img\s+([^>]+)>', sub_img, content, flags=re.DOTALL)
    
    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

for f in files_to_check:
    fix_file(f)

